File: schedule-service/scripts/seed_macro_zones.py
import json, os, psycopg2
from dotenv import load_dotenv

load_dotenv()
DB_URL = os.getenv('DB_URL')

# Zones macro data
zones_data = [
    {
        "zone_name": "URBANO_CENTRAL",
        "route_name": "URBANO_CENTRAL",
        "schedule_day": 0,  # All days
        "schedule_config": "NOCTURNO (21:00)",
        "geometry": {
            "type": "MultiPolygon",
            "coordinates": [[[[
                [-78.625, -0.925],
                [-78.61, -0.925],
                [-78.61, -0.945],
                [-78.625, -0.945],
                [-78.625, -0.925]
            ]]]]
        }
    },
    {
        "zone_name": "URBANO_NORTE",
        "route_name": "URBANO_NORTE",
        "schedule_day": 2,  # Tuesday
        "schedule_config": "DIURNO (Mar-Jue-Sab)",
        "geometry": {
            "type": "MultiPolygon",
            "coordinates": [[[[
                [-78.625, -0.925],
                [-78.61, -0.925],
                [-78.6, -0.89],
                [-78.63, -0.89],
                [-78.625, -0.925]
            ]]]]
        }
    },
    {
        "zone_name": "URBANO_SUR",
        "route_name": "URBANO_SUR",
        "schedule_day": 1,  # Monday
        "schedule_config": "DIURNO (Lun-Mie-Vie)",
        "geometry": {
            "type": "MultiPolygon",
            "coordinates": [[[[
                [-78.625, -0.945],
                [-78.61, -0.945],
                [-78.6, -0.98],
                [-78.64, -0.98],
                [-78.625, -0.945]
            ]]]]
        }
    },
    {
        "zone_name": "RURAL_NORTE",
        "route_name": "RURAL_NORTE",
        "schedule_day": 3,  # Wednesday
        "schedule_config": "RURAL (Rutas 2,3,4)",
        "geometry": {
            "type": "MultiPolygon",
            "coordinates": [[[[
                [-78.63, -0.89],
                [-78.6, -0.89],
                [-78.55, -0.8],
                [-78.7, -0.8],
                [-78.63, -0.89]
            ]]]]
        }
    },
    {
        "zone_name": "RURAL_SUR",
        "route_name": "RURAL_SUR",
        "schedule_day": 5,  # Friday
        "schedule_config": "RURAL (Rutas 1,5)",
        "geometry": {
            "type": "MultiPolygon",
            "coordinates": [[[[
                [-78.64, -0.98],
                [-78.6, -0.98],
                [-78.55, -1.05],
                [-78.7, -1.05],
                [-78.64, -0.98]
            ]]]]
        }
    }
]

print(f'🔌 Connecting to database...')
conn = psycopg2.connect(DB_URL)
cur = conn.cursor()

print(f'🗑️  Clearing existing zones...')
cur.execute('DELETE FROM cleaning_zones')

print(f'📝 Inserting {len(zones_data)} macro zones...')
for zone in zones_data:
    geom_json = json.dumps(zone['geometry'])
    cur.execute('''
        INSERT INTO cleaning_zones(zone_name, route_name, schedule_day, schedule_config, points_count, geom, status)
        VALUES (%s, %s, %s, %s, %s, ST_GeomFromGeoJSON(%s), 'ACUMULANDO')
    ''', (zone['zone_name'], zone['route_name'], zone['schedule_day'], zone['schedule_config'], 4, geom_json))
    print(f'  ✅ {zone["zone_name"]}')

conn.commit()
cur.close()
conn.close()

print(f'\n✅ Successfully inserted {len(zones_data)} macro zones!')

File: schedule-service/scripts/seed_macro_zones.go
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/joho/godotenv"
	"gorm.io/driver/postgres"
	"gorm.io/gorm"
)

type ZoneData struct {
	Name           string
	Route          string
	Day            int
	ScheduleConfig string
	Geom           string
}

func main() {
	// Load .env from parent directory
	if err := godotenv.Load("../.env"); err != nil {
		log.Println("Warning: .env not found, using environment variables")
	}

	dbURL := os.Getenv("DB_URL")
	if dbURL == "" {
		log.Fatal("DB_URL environment variable not set")
	}

	db, err := gorm.Open(postgres.Open(dbURL), &gorm.Config{})
	if err != nil {
		log.Fatal(err)
	}
	sqlDB, _ := db.DB()
	defer sqlDB.Close()

	zones := []ZoneData{
		{
			Name:           "URBANO_CENTRAL",
			Route:          "URBANO_CENTRAL",
			Day:            0,
			ScheduleConfig: "NOCTURNO (21:00)",
			Geom:           `{"type":"MultiPolygon","coordinates":[[[[-78.625,-0.925],[-78.61,-0.925],[-78.61,-0.945],[-78.625,-0.945],[-78.625,-0.925]]]]}`,
		},
		{
			Name:           "URBANO_NORTE",
			Route:          "URBANO_NORTE",
			Day:            2,
			ScheduleConfig: "DIURNO (Mar-Jue-Sab)",
			Geom:           `{"type":"MultiPolygon","coordinates":[[[[-78.625,-0.925],[-78.61,-0.925],[-78.6,-0.89],[-78.63,-0.89],[-78.625,-0.925]]]]}`,
		},
		{
			Name:           "URBANO_SUR",
			Route:          "URBANO_SUR",
			Day:            1,
			ScheduleConfig: "DIURNO (Lun-Mie-Vie)",
			Geom:           `{"type":"MultiPolygon","coordinates":[[[[-78.625,-0.945],[-78.61,-0.945],[-78.6,-0.98],[-78.64,-0.98],[-78.625,-0.945]]]]}`,
		},
		{
			Name:           "RURAL_NORTE",
			Route:          "RURAL_NORTE",
			Day:            3,
			ScheduleConfig: "RURAL (Rutas 2,3,4)",
			Geom:           `{"type":"MultiPolygon","coordinates":[[[[-78.63,-0.89],[-78.6,-0.89],[-78.55,-0.8],[-78.7,-0.8],[-78.63,-0.89]]]]}`,
		},
		{
			Name:           "RURAL_SUR",
			Route:          "RURAL_SUR",
			Day:            5,
			ScheduleConfig: "RURAL (Rutas 1,5)",
			Geom:           `{"type":"MultiPolygon","coordinates":[[[[-78.64,-0.98],[-78.6,-0.98],[-78.55,-1.05],[-78.7,-1.05],[-78.64,-0.98]]]]}`,
		},
	}

	// Verificar si ya existen zonas
	var count int64
	db.Raw("SELECT COUNT(*) FROM cleaning_zones").Scan(&count)

	if count > 0 {
		fmt.Printf("✅ Ya existen %d zonas en la base de datos. No se insertarán duplicados.\n", count)
		fmt.Println("💡 Si deseas recargar las zonas, ejecuta primero: DELETE FROM cleaning_zones;")
		return
	}

	fmt.Printf("📝 Insertando %d macro zones...\n", len(zones))
	for _, zone := range zones {
		err := db.Exec(`
			INSERT INTO cleaning_zones(zone_name, route_name, schedule_day, schedule_config, points_count, geom, status)
			VALUES ($1, $2, $3, $4, 4, ST_GeomFromGeoJSON($5), 'ACUMULANDO')
			ON CONFLICT (route_name, schedule_day) DO NOTHING
		`, zone.Name, zone.Route, zone.Day, zone.ScheduleConfig, zone.Geom).Error
		if err != nil {
			log.Printf("❌ Failed to insert %s: %v", zone.Name, err)
		} else {
			fmt.Printf("  ✅ %s\n", zone.Name)
		}
	}

	db.Raw("SELECT COUNT(*) FROM cleaning_zones").Scan(&count)
	fmt.Printf("\n✅ Total de zonas en la base de datos: %d\n", count)
}
